backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,186 @@
+
+            string username = mqttConfig.GetValue<string>("Username");
+            string password = config.GetValue<string>("mqtt-broker-password");
+
+            var builder = new MqttClientOptionsBuilder()
+                .WithTcpServer(
+                    mqttConfig.GetValue<string>("Host"),
+                    mqttConfig.GetValue<int>("Port")
+                )
+                .WithCredentials(username, password);
+
+            _options = new ManagedMqttClientOptionsBuilder()
+                .WithAutoReconnectDelay(_reconnectDelay)
+                .WithClientOptions(builder.Build())
+                .Build();
+
+            RegisterCallbacks();
+
+            var topics = mqttConfig.GetSection("Topics").Get<List<string>>();
+            SubscribeToTopics(topics);
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            _logger.LogInformation("MQTT client STARTED");
+            await _mqttClient.StartAsync(_options);
+            await stoppingToken;
+            await _mqttClient.StopAsync();
+            _logger.LogInformation("MQTT client STOPPED");
+        }
+
+        /// <summary>
+        /// The callback function for when a subscribed topic publishes a message
+        /// </summary>
+        /// <param name="messageReceivedEvent"> The event information for the MQTT message </param>
+        private void OnMessageReceived(MqttApplicationMessageReceivedEventArgs messageReceivedEvent)
+        {
+            string content = messageReceivedEvent.ApplicationMessage.ConvertPayloadToString();
+            string topic = messageReceivedEvent.ApplicationMessage.Topic;
+
+            var messageType = MqttTopics.TopicsToMessages.GetItemByTopic(topic);
+            if (messageType is null)
+            {
+                _logger.LogError("No message class defined for topic '{topicName}'", topic);
+                return;
+            }
+
+            _logger.LogDebug("Topic: {topic} - Message recieved: \n{payload}", topic, content);
+
+            switch (messageType)
+            {
+                case Type type when type == typeof(IsarMission):
+                    OnIsarTopicReceived<IsarMission>(content);
+                    break;
+                case Type type when type == typeof(IsarTask):
+                    OnIsarTopicReceived<IsarTask>(content);
+                    break;
+                case Type type when type == typeof(IsarStep):
+                    OnIsarTopicReceived<IsarStep>(content);
+                    break;
+                default:
+                    _logger.LogWarning(
+                        "No callback defined for MQTT message type '{type}'",
+                        messageType.Name
+                    );
+                    break;
+            }
+        }
+
+        private void OnConnected(MqttClientConnectedEventArgs obj)
+        {
+            _logger.LogInformation("Successfully connected.");
+        }
```

Comment:
I would suggest including some more information in this log message just because it can be useful. For example adding which host and port you have connected to.

Reply:
I assume that may be found in the `MqttClientConnectedEventArgs`

Reply:
Agreed